### PR TITLE
Chronos: support customized loss and metric in tf forecaster

### PR DESCRIPTION
**Pull request by Chaselzxy**

### CODE REVIEW

File: python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py [hunk 1]
```diff
@@ -95,6 +95,31 @@ def test_lstm_forecaster_save_load(self):
...
+            return tf.losses.mean_squared_error(tf.convert_to_tensor(y_pred),
+                                      tf.convert_to_tensor(y_true)).numpy()
+        from bigdl.chronos.forecaster.tf.lstm_forecaster import LSTMForecaster
+        self.forecaster = LSTMForecaster(past_seq_len=10,
+                                    input_feature_num=10,
+                                    output_feature_num=2,
+                                    loss=loss,
+                                    metrics=[customized_metric],
+                                    lr=0.01)
+        self.forecaster.fit(train_data, epochs=2, batch_size=32)
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:112]: oh so this means that we can directly support customized loss and metric?

Comment (Chaselzxy) [reply python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:112]: > 

yes

Comment (Chaselzxy) [reply python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:112]: > oh so this means that we can directly support customized loss and metric?

yes

Comment (TheaperDeng) [reply python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:112]: that's great.

File: python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py [hunk 2]
```diff
@@ -95,6 +95,31 @@ def test_lstm_forecaster_save_load(self):
         load_model_yhat = self.forecaster.predict(test_data[0])
         assert yhat.shape == (400, 1, 2)
         np.testing.assert_almost_equal(yhat, load_model_yhat, decimal=5)
+    
+    def test_lstm_customized_loss_metric(self):
+        train_data, test_data = create_data(tf_data=True)
+        loss = tf.losses.mean_squared_error
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:101]: Well I am not quite sure if we need to make this an tf2 loss? `tf.keras.losses.MeanSquaredError()`
reference: https://www.tensorflow.org/api_docs/python/tf/compat/v1/losses/mean_squared_error

File: python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py [hunk 3]
```diff
@@ -62,14 +62,18 @@ def __init__(self,
                (i.e. the close possibility to a neuron). This value defaults to 0.1.
         :param optimizer: Specify the optimizer used for training. This value
                defaults to "Adam".
-        :param loss: Specify the loss function used for training. This value
-               defaults to "mse". You can choose from "mse", "mae" and
-               "huber_loss".
+        :param loss: Str or tensorflow loss instance, specify the loss function
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py:65]: Str or a tf.keras.losses.Loss instance

File: python/chronos/test/bigdl/chronos/forecaster/tf/test_seq2seq_keras_forecaster.py [hunk 4]
```diff
@@ -96,6 +95,31 @@ def test_seq2seq_save_load(self):
...
+            self.forecaster.load(tmp_dir_file)
+            from bigdl.chronos.model.tf2.Seq2Seq_keras import LSTMSeq2Seq
+            assert isinstance(self.forecaster.internal, LSTMSeq2Seq)
+        load_model_yhat = self.forecaster.predict(test_data)
+        assert yhat.shape == (400, 2, 2)
+        np.testing.assert_almost_equal(yhat, load_model_yhat, decimal=5)
 
 if __name__ == '__main__':
-    pytest.main([__file__])
+    pytest.main([__file__])
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/tf/test_seq2seq_keras_forecaster.py:125]: need a blank line after this line

**Comment by Chaselzxy** [comment]
http://10.112.231.51:18888/job/Chronos-PR-Validation-Spark-3.1/1097/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-onnxrt1.6/942/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-ray-part1/1114/

**Comment by TheaperDeng** [comment]
> http://10.112.231.51:18888/job/Chronos-PR-Validation-Spark-3.1/1097/ http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-onnxrt1.6/942/ http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-ray-part1/1114/

These are only part of the tests: `BigDL-Chronos-PR-Validation` is needed.

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18889/job/BigDL-Chronos-PR-Validation/520/
